superset-frontend/src/explore/components/DataTablesPane/index.tsx

```diff
@@ -192,7 +250,25 @@ export const DataTablesPane = ({
               [resultType]: json.result[0].data,
             }));
           }
-
+          const checkCols = json?.result[0]?.data?.length
+            ? Object.keys(json.result[0].data[0])
+            : null;
+          if (!json.result[0].colnames && checkCols?.length) {
+            let tabKey: string = RESULT_TYPES.results;
+            if (resultType === RESULT_TYPES.samples) {
```

Comment:
I'm wondering if we could simplify the logic here? For example, `tabKey` variable seems redundant. Also, I'm not sure if `resultType === RESULT_TYPES.samples` check is necessary.
Can't we do this instead?
```
setColumnNames(prevColumnNames => ({
              ...prevColumnNames,
              [resultType]: json.result[0].colnames || checkCols,
            }));
```